### PR TITLE
Product entities

### PR DESCRIPTION
**Pull request by dpolant**
Product/product type entity UI and field definitions

### CODE REVIEW

File: modules/product/commerce_product.local_actions.yml [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+  route_name: commerce_product.product_add_page
+  title: 'Add a new product'
+  appears_on:
+    - commerce_product.list
+
+commerce_product.product_type_add:
+  route_name: commerce_product.product_type_add
+  title: 'Add a new product type'
+  appears_on:
+    - commerce_product.product_type_list
```
Comment (bojanz) [inline modules/product/commerce_product.local_actions.yml]: Missing new line.

File: modules/product/commerce_product.module [hunk 2]
```diff
@@ -0,0 +1,35 @@
+<?php
+
+use Drupal\commerce_product\Entity\CommerceProductType;
+
+/**
+ * @file
+ * Defines the core Commerce product entity, including the entity itself, the
+ * bundle definitions (product types), and various API functions to manage
+ * products and interact with them through forms and autocompletes.
+ */
```
Comment (bojanz) [inline modules/product/commerce_product.module:10]: Missing empty line between two docblocks.

File: modules/product/src/CommerceProductInterface.php [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+ * Provides an interface defining a Commerce Product entity.
+ */
+interface CommerceProductInterface extends EntityInterface {
+  /**
+   * Returns the identifier.
+   *
+   * @return int
+   *   The entity identifier.
+   */
+  public function id();
```
Comment (bojanz) [inline modules/product/src/CommerceProductInterface.php]: The interface should not contain methods that already exist on the parent interface.
On the other hand, it must contain methods specific to this entity class (like the sku getter and setter, etc)

File: modules/product/src/Entity/CommerceProduct.php [hunk 4]
```diff
@@ -0,0 +1,261 @@
...
+ *   label = @Translation("Product"),
+ *   controllers = {
+ *     "view_builder" = "Drupal\Core\Entity\EntityViewBuilder",
+ *     "list_builder" = "Drupal\commerce_product\CommerceProductListBuilder",
+ *     "form" = {
+ *       "add" = "Drupal\commerce_product\Form\CommerceProductForm",
+ *       "edit" = "Drupal\commerce_product\Form\CommerceProductForm",
+ *       "delete" = "Drupal\commerce_product\Form\CommerceProductDeleteForm"
+ *     },
+ *     "translation" = "Drupal\content_translation\ContentTranslationController"
```
Comment (bojanz) [inline modules/product/src/Entity/CommerceProduct.php]: according to latest core code, this is supposed to be
"translation" = "Drupal\content_translation\ContentTranslationHandler"

File: modules/product/src/Entity/CommerceProduct.php [hunk 5]
```diff
@@ -0,0 +1,261 @@
...
+  public function setSku($sku) {
+    $this->set('sku', $sku);
+    return $this;
+  }
+  
+  /**
+   * {@inheritdoc}
+   */
+  public function getRevisionId() {
+    return $this->get('revision_id')->value;
```
Comment (bojanz) [inline modules/product/src/Entity/CommerceProduct.php]: Remove id() and getRevisionId(), they already exist on the parent and work the same.
And there is no point in having setRevisionId(), that id is not changeable.

File: modules/product/src/Form/CommerceProductForm.php [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+      $form['status']['widget']['#default_value'] = 1;
+    }
+    
+    return $form;
+  }
+
+  /**
+   * Overrides \Drupal\Core\Entity\EntityFormController::submit().
+   */
+
```
Comment (bojanz) [inline modules/product/src/Form/CommerceProductForm.php]: Extra newline.

File: modules/product/src/CommerceProductInterface.php (hunk 3 above)
Comment (rszrama) [inline modules/product/src/CommerceProductInterface.php]: Guess I should fix this in Store, too; I didn't double check it, but the tutorial was wrong on this point.

File: modules/product/src/Entity/CommerceProduct.php (hunk 4 above)
Comment (rszrama) [inline modules/product/src/Entity/CommerceProduct.php]: D'oh, this is wrong in Store as well. :-/

**Pull request merged**